### PR TITLE
[Microsoft.ApiManagement][2019-12-01] Gateway hostname configuration protocol settings

### PR DESCRIPTION
**Pull request by maksimkim**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Review by solankisamir — APPROVED** [review]

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2019-12-01/definitions.json [hunk 1]
```diff
@@ -5525,6 +5525,18 @@
         "negotiateClientCertificate": {
           "type": "boolean",
           "description": "Determines whether gateway requests client certificate"
+        },
```
Comment (KrisBash) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2019-12-01/definitions.json:5528]: Adding these within an API version can cause undesired effects as customers use mixed SDK versions where only latest can deserialize these properties. Additionally, Azure Policy becomes unpredictable if properties customers use in policy evaluations are added within an API version. Can they be added in a new version?

**Review by KrisBash — CHANGES_REQUESTED** [review]
Please see one comment

**Comment by ravbhatnagar** [comment]
@solankisamir - please let us know if this PR is still relevant and re-add the ARM review label. removing from now due to no activity for several months.

**Comment by solankisamir** [comment]
@ravbhatnagar we have put the changes into new PR https://github.com/Azure/azure-rest-api-specs/pull/10292

**Pull request closed without merging**